libs/mockito-extensions/src/main/java/org/codice/ddf/test/mockito/PrivilegedVerificationMode.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.test.mockito;
+
+import static org.mockito.Mockito.times;
+
+import java.lang.reflect.Field;
+import java.security.AccessController;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+import org.mockito.Incubating;
+import org.mockito.exceptions.base.MockitoAssertionError;
+import org.mockito.internal.debugging.LocationImpl;
+import org.mockito.internal.verification.VerificationModeFactory;
+import org.mockito.internal.verification.api.VerificationData;
+import org.mockito.internal.verification.api.VerificationDataInOrder;
+import org.mockito.internal.verification.api.VerificationInOrderMode;
+import org.mockito.invocation.Invocation;
+import org.mockito.invocation.Location;
+import org.mockito.invocation.MatchableInvocation;
+import org.mockito.verification.VerificationMode;
+
+/**
+ * Class that extends Mockito's {@link VerificationMode} to verify that a mocked method has been
+ * called within a {@code AccessController#doPrivileged()} block.
+ *
+ * <p><b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+@Incubating
+public class PrivilegedVerificationMode implements VerificationMode, VerificationInOrderMode {
+
+  private final VerificationMode mode;
+
+  private PrivilegedVerificationMode(VerificationMode mode) {
+    this.mode = mode;
+  }
+
+  /**
+   * Wraps another {@link VerificationMode} instance and ensures all {@code verify()} calls from the
+   * specified mode were made inside a {@code AccessController.doPrivileged()} block.
+   *
+   * @param mode {@link VerificationMode} instance to wrap with a privileged check
+   * @return wrapped {@link VerificationMode} instance
+   */
+  public static VerificationMode privileged(VerificationMode mode) {
```

Comment:
❓ I don't know if it matters, but would it make more sense for these statics to be overloaded constructors instead, since all they do is return an instance of the enclosing class?

Reply:
This is mostly to be consistent with the way Mockito already does this, i.e., they provide a static `times()` method that returns a `VerificationMode` instead of having you instantiate the class itself.